service/controller/clusterapi/v16/key/cluster.go

```diff
@@ -1,14 +1,67 @@
 package key
 
 import (
+	"github.com/giantswarm/cluster-operator/pkg/label"
 	"github.com/giantswarm/microerror"
 	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
+func AWSClusterConfigName(cluster v1alpha1.Cluster) string {
+	return clusterProviderStatus(cluster).Cluster.ID
+}
+
+func ClusterAvailabilityZones(cluster v1alpha1.Cluster, machineDeployments []v1alpha1.MachineDeployment) []string {
+	azMap := make(map[string]struct{})
+
+	azMap[ClusterMasterAZ(cluster)] = struct{}{}
+
+	// TODO: Extract AZs from MachineDeployments
+
+	azs := make([]string, 0, len(azMap))
+	for az := range azMap {
+		azs = append(azs, az)
+	}
+	return azs
+}
+
+func ClusterCredentialSecretName(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Provider.CredentialSecret.Name
+}
+
+func ClusterCredentialSecretNamespace(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Provider.CredentialSecret.Namespace
+}
+
+func ClusterDNSZone(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Cluster.DNS.Domain
+}
+
 func ClusterID(cluster v1alpha1.Cluster) string {
 	return clusterProviderStatus(cluster).Cluster.ID
 }
 
+func ClusterMasterAZ(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Provider.Master.AvailabilityZone
+}
+
+func ClusterMasterInstanceType(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Provider.Master.InstanceType
+}
+
+func ClusterName(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).ClusterName
+}
+
+func ClusterReleaseVersion(cluster v1alpha1.Cluster) string {
+	relVer, ok := cluster.Labels[label.ReleaseKey]
+	if !ok {
+		// Later on we will have admission controller validating Cluster object
+		// so we shall not hit this panic.
```

Comment:
This won't be even possible to observe an object w/o this set because we'll have label selectors like `release.giantswarm.io/version=1.2.3` for aws-operator instance handling version 1.2.3.